### PR TITLE
Update shadow resource to use FilterTable

### PR DESCRIPTION
**Pull request by miah**
This PR updates the shadow resource to use FilterTable rather than a custom filter implementation.
I've also increased test coverage =)

Fixes https://github.com/chef/inspec/issues/2265

### CODE REVIEW

**Review by jquick — APPROVED** [review]
Looks good! Thanks @miah

File: lib/resources/shadow.rb [hunk 1]
```diff
@@ -32,87 +30,55 @@ class Shadow < Inspec.resource(1)
...
+
+    filter.add(:users) { |u, _| u.entries.map(&:user) }
+
+    filter.add(:count) { |i, _|
+      i.entries.length
+    }
+
+    filter.connect(self, :params)
+
+    alias users user
```
Comment (chris-rock) [inline lib/resources/shadow.rb]: Should we add some deprecations here? Based on that improvement, I am not sure what you think users should use.

**Review by chris-rock — COMMENTED** [review]
Huge improvements. The only part that is missing to me is a clear statement how the resource should be used. This would require to update the examples, docs and highlight deprecations for InSpec 3.0

File: lib/resources/shadow.rb (hunk 1 above)
Comment (miah) [reply lib/resources/shadow.rb]: Yes, I think we should deprecate the singular forms of user and password. I can update this PR with deprecation notices for those targeting 3.0. I'll update the docs too =)

**Review by chris-rock — APPROVED** [review]
Great improvement. Thank you @miah

**Pull request merged**